### ISSUE TITLE
`Blob::new(&[u8])` fails when wasm memory is backed by `SharedArrayBuffer`

### ISSUE THREAD

**Issue by Liamolucko** [rustwasm/gloo#208]
### Describe the Bug

I've just stumbled upon an issue with my changes in #152, which is that `new Blob()` doesn't allow passing `Uint8Array`s backed by `SharedArrayBuffer`. This is problematic because `SharedArrayBuffer` is sometimes used to back wasm memory, which will then cause `Blob::new(&[u8])` to fail.

This isn't an issue for `Blob::new(ArrayBuffer)`, since `SharedArrayBuffer` isn't actually a subclass of `ArrayBuffer`.

`Blob::new` will have to fall back to copying the buffer when wasm memory is backed by a `SharedArrayBuffer`, or just do it unconditionally.

### Steps to Reproduce

1. Clone https://github.com/Liamolucko/gloo-sab-repro
2. Build and bindgen it:
```sh
cargo build --release
wasm-bindgen target/wasm32-unknown-unknown/release/gloo-sab-repro.wasm --out-dir dist --target web
```
3. Run the resulting wasm in a web browser or Deno

### Expected Behavior

It should run successfully.

### Actual Behavior

`new Blob()` throws an error, which causes a panic since it's not expected to fail.

### Additional Context

This is actually the case (according to the spec, at least) for basically everything accepting a buffer source, because [WebIDL doesn't consider `SharedArrayBuffer` a real buffer source unless the type has the `[AllowShared]` attribute](https://webidl.spec.whatwg.org/#es-buffer-source-types). Based on a quick grep of web-sys's WebIDL definitions, only WebGPU is using that right now.

I found out about this from `getrandom`'s similar issue: rust-random/getrandom#164.

**Comment by hamza1311** [comment]
Do you know of a fix for this?

**Comment by Liamolucko** [comment]
Yes; if the wasm memory is backed by a `SharedArrayBuffer`, the bytes will have to be copied into a regular `ArrayBuffer` before passing to `new Blob()` (probably via. `Uint8Array::from`). It's a bit annoying having to re-introduce an intermediate copy, but it's more important that it works.
